### ISSUE TITLE
GCC 11 throws `mismatched bound` errors on arm64

### ISSUE THREAD

**Issue by merkys** [open-quantum-safe/liboqs#1125]
I am interested in packaging liboqs for Debian. Debian build of ba5b61a7 on arm64 fails due to `mismatched bound` errors:
```
/<<PKGBUILDDIR>>/src/kem/kyber/pqclean_kyber512_aarch64/kem.c:24:57: error: argument 1 of type ‘uint8_t[800]’ {aka ‘unsigned char[800]’} with mismatched bound [-Werror=array-parameter=]
   24 | int PQCLEAN_KYBER512_AARCH64_crypto_kem_keypair(uint8_t pk[KYBER_PUBLICKEYBYTES],
      |                                                 ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /<<PKGBUILDDIR>>/src/kem/kyber/pqclean_kyber512_aarch64/kem.c:2:
/<<PKGBUILDDIR>>/src/kem/kyber/pqclean_kyber512_aarch64/kem.h:14:58: note: previously declared as ‘uint8_t *’ {aka ‘unsigned char *’}
   14 | int PQCLEAN_KYBER512_AARCH64_crypto_kem_keypair(uint8_t *pk, uint8_t *sk);
      |                                                 ~~~~~~~~~^~
```
([full build log](https://buildd.debian.org/status/fetch.php?pkg=liboqs&arch=arm64&ver=0.7.0.27.gba5b61a7%2Bdfsg-1&stamp=1637141594&raw=0); note Debian turns all GCC warning to errors)

Similar issue was reported as #1035 and seemingly has been fixed since then. Interestingly, most of the other architectures build just fine.

**Comment by baentsch** [comment]
Thanks for the notification. We recently added aarch64-optimized code, so this may indeed be a regression: While we look into your report, would you want to check your build based on a commit prior to #1117?

**Comment by bhess** [comment]
The aarch64 code is pulled from pqclean while the other versions from pqcrystals-upstream. It seems that the pqclean-version needs to be updated with the recent fix for GCC 11 available on pqcrystals-upstream. See https://github.com/pq-crystals/kyber/pull/43.

**Comment by merkys** [comment]
> Thanks for the notification. We recently added aarch64-optimized code, so this may indeed be a regression: While we look into your report, would you want to check your build based on a commit prior to #1117?

OK, I will try building it.

**Comment by merkys** [comment]
I can confirm that c366e19805387d70483c460b77afb10b7c9c9269 (commit before merging #1117) builds just fine on aarch64.

**Comment by baentsch** [comment]
So maybe a question to @jschanck : Would addressing https://github.com/PQClean/PQClean/issues/420#issuecomment-971547997 be "quick and easy" for you (maybe extending https://github.com/PQClean/PQClean/pull/426)? 

If so, we'd pull it in when closing out #1031. Otherwise, what about a patch to Kyber-aarch64 as part of https://github.com/open-quantum-safe/liboqs/pull/1123 to resolve this issue before the `liboqs` release as a solution until the PQClean-aarch64-Kyber code gets upgraded? Thoughts, @Martyrshot @dstebila ?

**Comment by jschanck** [comment]
Matthias has been packaging the aarch64 version. I tagged him in the PQClean issue.

**Comment by mkannwischer** [comment]
I've fixed it upstream: https://github.com/neon-ntt/neon-ntt/commit/c6ffbcd1e5590e4dddbc604fc80d2ad756485b4d
I also opened a PR to PQClean: https://github.com/PQClean/PQClean/pull/429

**Comment by dstebila** [comment]
> I've fixed it upstream: [neon-ntt/neon-ntt@c6ffbcd](https://github.com/neon-ntt/neon-ntt/commit/c6ffbcd1e5590e4dddbc604fc80d2ad756485b4d) I also opened a PR to PQClean: [PQClean/PQClean#429](https://github.com/PQClean/PQClean/pull/429)

Thanks Matthias! I've reviewed and merged them. I'll create a PR for liboqs.
